tests/src/test/scala/cats/tests/MonadTest.scala

```diff
@@ -74,4 +74,28 @@ class MonadTest extends CatsSuite {
     result should ===(50000)
   }
 
+  test("iterateWhileM") {
+    forAll(Gen.posNum[Int]) { (max: Int) =>
```

Comment:
might this not run a super long time? any reason why `max` shouldn't be something from like: `Gen.choose(0, 500000)`?